flytekit/core/python_function_task.py

```diff
@@ -159,7 +159,79 @@ def istestfunction(func) -> bool:
     return False
 
 
-class PythonFunctionTask(PythonAutoContainerTask[T]):
+class TaskLoader(object):
+    """
+    A TaskLoader that can be used to load the task itself from the actual argument that is captured.
+    The argument itself should be discoverable through the class loading framework.
+    """
+
+    @abstractmethod
+    def load_task(self, loader_args: List[str]) -> "PythonInstanceTask":
+        pass
+
+    @abstractmethod
+    def loader_args(self, var: InstanceVar) -> List[str]:
+        pass
+
+
+class PythonInstanceTask(PythonAutoContainerTask[T], ABC):
+    """
+    This class should be used as the base class for all Tasks that do not have a user defined function body, but have
+    a platform defined execute method. (Execute needs to be overriden). This base class ensures that the module loader
+    will invoke the right class automatically, by capturing the module name and variable in the module name.
+
+    .. code-block: python
+
+        x = MyInstanceTask(name="x", .....)
+
+        # this can be invoked as
+        x(a=5) # depending on the interface of the defined task
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        task_config: T,
+        task_type: str = "python-task",
+        task_loader: Optional[TaskLoader] = None,
+        **kwargs,
+    ):
+        super().__init__(name=name, task_config=task_config, task_type=task_type, **kwargs)
+        self._task_loader = task_loader
+
+    @property
+    def task_loader(self):
+        return self._task_loader
+
+    def get_command(self, settings: SerializationSettings) -> List[str]:
+        """
+        NOTE: This command is different, it tries to retrieve the actual LHS of where this object was assigned, so that
+        the module loader can easily retreive this for execution - at runtime.
+        """
+        var = settings.get_instance_var(self)
```

Comment:
Should this be
```suggestion
        var = settings.get_instance_var(self.task_loader)
```

? or am I missing something?

Reply:
ya it should be you are right, if task_loader is set